[Emscripten] testcontroller error and checkkeys warning

I tried compiling SDL with Emscripten following the `README-emscripten.md` documentation.

Building SDL and examples work.

But building the tests fails:

```c
[ 50%] Building C object test/CMakeFiles/checkkeys.dir/checkkeys.c.o
/path/to/SDL-git/test/checkkeys.c:237:17: warning: incompatible pointer types initializing 'const bool *' with an expression of type 'const unsigned char *' [-Wincompatible-pointer-types]
  237 |     const bool *keystate = SDL_GetKeyboardState(&max_keys);
      |                 ^          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```c
[ 75%] Building C object test/CMakeFiles/testcontroller.dir/testcontroller.c.o
/path/to/SDL-git/test/testcontroller.c:1175:17: error: incompatible function pointer types assigning to 'unsigned char (*)(void *, Uint16, Uint16)' (aka 'unsigned char (*)(void *, unsigned short, unsigned short)') from 'bool (void *, Uint16, Uint16)' (aka 'bool (void *, unsigned short, unsigned short)') [-Wincompatible-function-pointer-types]
 1175 |     desc.Rumble = VirtualGamepadRumble;
      |                 ^ ~~~~~~~~~~~~~~~~~~~~
/path/to/SDL-git/test/testcontroller.c:1176:25: error: incompatible function pointer types assigning to 'unsigned char (*)(void *, Uint16, Uint16)' (aka 'unsigned char (*)(void *, unsigned short, unsigned short)') from 'bool (void *, Uint16, Uint16)' (aka 'bool (void *, unsigned short, unsigned short)') [-Wincompatible-function-pointer-types]
 1176 |     desc.RumbleTriggers = VirtualGamepadRumbleTriggers;
      |                         ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/path/to/SDL-git/test/testcontroller.c:1177:17: error: incompatible function pointer types assigning to 'unsigned char (*)(void *, Uint8, Uint8, Uint8)' (aka 'unsigned char (*)(void *, unsigned char, unsigned char, unsigned char)') from 'bool (void *, Uint8, Uint8, Uint8)' (aka 'bool (void *, unsigned char, unsigned char, unsigned char)') [-Wincompatible-function-pointer-types]
 1177 |     desc.SetLED = VirtualGamepadSetLED;
      |                 ^ ~~~~~~~~~~~~~~~~~~~~
```

Because of the `bool`/`unsigned char` mismatch, my guess is that SDL was built with C, but the tests are built with C++.

> /path/to/SDL-git/test/checkkeys.c:237:17: warning: incompatible pointer types initializing 'const bool *' with an expression of type 'const unsigned char *' [-Wincompatible-pointer-types]


Do these warnings still appear? This looks like `bool` was `#define`'d as `unsigned char` in the SDL headers.
And later, `bool` was typedef'ed (probably through `<stdbool.h>` or from `_Bool`).

`SDL_keyboard.h` from current main includes `SDL_stdinc.h` so `bool` should be a proper type, and not a macro.

I updated my emsdk, and I can reproduce your warning:
```
/home/maarten/projects/SDL/test/checkkeys.c:237:17: warning: incompatible pointer types initializing 'const bool *' with an expression of type 'const unsigned char *' [-Wincompatible-pointer-types]
  237 |     const bool *keystate = SDL_GetKeyboardState(&max_keys);
      |                 ^          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

--- a/include/SDL3/SDL_stdinc.h
+++ b/include/SDL3/SDL_stdinc.h
@@ -45,6 +45,11 @@
 #endif
 
 #ifndef __cplusplus
+#if defined(__has_include) && !defined(SDL_INCLUDE_STDBOOL_H)
+#if __has_include(<stdbool.h>)
+#define SDL_INCLUDE_STDBOOL_H
+#endif
+#endif
 #if (defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L) || \
     (defined(_MSC_VER) && (_MSC_VER >= 1910 /* Visual Studio 2017 */)) || \
     defined(SDL_INCLUDE_STDBOOL_H)

